src/protocol-parser/protocol-parser.ts

```diff
@@ -744,6 +746,68 @@ export namespace BrowsingContext {
     };
   };
 
+  export type PrintCommand = {
+    method: 'browsingContext.print';
+    params: PrintParameters;
+  };
+
+  // All units are in cm.
+  const PrintPageSchema = zod.object({
+    height: zod.number().min(0.0).default(27.94).optional(),
+    width: zod.number().min(0.0).default(21.59).optional(),
+  });
+
+  // All units are in cm.
+  const PrintMarginSchema = zod.object({
+    bottom: zod.number().min(0.0).default(1.0).optional(),
+    left: zod.number().min(0.0).default(1.0).optional(),
+    right: zod.number().min(0.0).default(1.0).optional(),
+    top: zod.number().min(0.0).default(1.0).optional(),
+  });
+
+  /**
+   * 1-based index, e.g. 1, '5', '2-4', '6-', '-3'
+   * @see https://w3c.github.io/webdriver/#dfn-parse-a-page-range
+   */
+  const PrintPageRangesSchema = zod
+    .array(
+      zod.union([
+        zod
+          .string()
+          .min(1)
+          .regex(/^((?:\d+)|(?:\d+[-])|(?:[-]\d+)|(?:\d+[-]\d+))$/),
```

Comment:
Can we use a preprocess here instead of regex?

Reply:
Array of integers:

- (1) is not expressive enough: for example, we cannot express both '1' and '3-5' with one
- (2) is not efficient: for example, if we printed '1-1000000' we would have to have an `[1,2,...,1000000]` array

That's how I read your suggestion; please correct me if you thought of another representation. String seems more flexible for what we're looking for, especially considering that the CDP param is also a string.

Reply:
Resolving. Accomplished with `.refine()` and `.transform()`.